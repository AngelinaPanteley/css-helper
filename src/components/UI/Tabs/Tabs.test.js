"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { expect as chaiExpect } from 'chai';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import MyTabs from './Tabs.js';
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';
configure({ adapter: new Adapter() });

describe('>>>Tabs', () => {

  //SNAPSHOTS

  const props = {
    tabs: ['tab1', 'tab2'],
  }

  const tabsComponent = <MyTabs {...props}>
    <div>Panel1</div>
    <div>Panel2</div>
  </MyTabs>
    ;

  const component = renderer.create(tabsComponent);

  it('renders itself correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });

  //SHALLOW RENDERING

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(tabsComponent);
  });

  it('renders Tabs', () => {
    chaiExpect(wrapper.find(Tabs)).to.have.lengthOf(1);
  });

  it('renders TabList', () => {
    chaiExpect(wrapper.find(TabList)).to.have.lengthOf(1);
  });

  it('renders Tab components', () => {
    chaiExpect(wrapper.find(Tab)).to.have.lengthOf(2);
  });

  it('renders TabPanel components', () => {
    chaiExpect(wrapper.find(TabPanel)).to.have.lengthOf(2);
  });

})