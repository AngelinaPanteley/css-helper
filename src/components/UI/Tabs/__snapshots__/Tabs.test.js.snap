// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Tabs renders itself correctly 1`] = `
<div
  className="Tabs"
  data-tabs={true}
  onClick={[Function]}
  onKeyDown={[Function]}
>
  <ul
    className="react-tabs__tab-list"
    role="tablist"
  >
    <li
      aria-controls="react-tabs-1"
      aria-disabled="false"
      aria-selected="true"
      className="react-tabs__tab react-tabs__tab--selected"
      id="react-tabs-0"
      role="tab"
      style={
        Object {
          "width": "calc(50% + 18px)",
        }
      }
      tabIndex="0"
    >
      tab1
    </li>
    <li
      aria-controls="react-tabs-3"
      aria-disabled="false"
      aria-selected="false"
      className="react-tabs__tab"
      id="react-tabs-2"
      role="tab"
      style={
        Object {
          "width": "calc(50% + 18px)",
        }
      }
      tabIndex={null}
    >
      tab2
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-0"
    className="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-1"
    role="tabpanel"
  >
    <div>
      Panel1
    </div>
  </div>
  <div
    aria-labelledby="react-tabs-2"
    className="react-tabs__tab-panel"
    id="react-tabs-3"
    role="tabpanel"
  />
</div>
`;
